File: source_funcs/go_transcoder/MINIMUM_SUM_PRODUCT_TWO_ARRAYS/MINIMUM_SUM_PRODUCT_TWO_ARRAYS.go
package main

import "math"

func f_gold(a []int, b []int, n int, k int) int {
	var (
		diff int = 0
		res  int = 0
		temp int
	)
	for i := int(0); i < n; i++ {
		var pro int = a[i] * b[i]
		res = res + pro
		if pro < 0 && b[i] < 0 {
			temp = (a[i] + k*2) * b[i]
		} else if pro < 0 && a[i] < 0 {
			temp = (a[i] - k*2) * b[i]
		} else if pro > 0 && a[i] < 0 {
			temp = (a[i] + k*2) * b[i]
		} else if pro > 0 && a[i] > 0 {
			temp = (a[i] - k*2) * b[i]
		}
		var d int = int(math.Abs(float64(pro - temp)))
		if d > diff {
			diff = d
		}
	}
	return res - diff
}

File: rust_funcs/c_transcoder/MINIMUM_SUM_PRODUCT_TWO_ARRAYS/MINIMUM_SUM_PRODUCT_TWO_ARRAYS.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


use std::cmp::Ordering;

fn f_gold(a: [i32;2], b: [i32;2], n: i32, k: i32) -> i32 {
    let mut diff = 0;
    let mut res = 0; 
    for i in 0..n {
        let pro: i32 = a[i as usize] * b[i as usize];
        res += pro;
        if pro < 0 && b[i as usize] < 0  {
            let temp = (a[i as usize] + 2 * k) * b[i as usize];
            diff = diff.max(temp.abs() - pro.abs());
        } else if pro < 0 && a[i as usize] < 0  {
            let temp = (a[i as usize] - 2 * k) * b[i as usize];
            diff = diff.max(temp.abs() - pro.abs());

        } else if pro > 0 && a[i as usize] < 0  {
            let temp = (a[i as usize] + 2 * k) * b[i as usize];
            diff = diff.max(temp.abs() - pro.abs());
        } else if pro > 0 && a[i as usize] > 0  {
            let temp = (a[i as usize] - 2 * k) * b[i as usize];
            diff = diff.max(temp.abs() - pro.abs());
        }
    }
    res - diff
}
